apps/etterlatte-saksbehandling-ui/client/src/components/nyoppgavebenk/Oppgavelista.tsx

```diff
@@ -1,13 +1,129 @@
-import { Table } from '@navikt/ds-react'
+import { Button, Select, Table } from '@navikt/ds-react'
 import { formaterStringDato } from '~utils/formattering'
 import { TildelSaksbehandler } from '~components/nyoppgavebenk/TildelSaksbehandler'
 import { RedigerSaksbehandler } from '~components/nyoppgavebenk/RedigerSaksbehandler'
 import { OppgaveDTOny } from '~shared/api/oppgaverny'
+import { useState } from 'react'
+import styled from 'styled-components'
+import {
+  EnhetFilterKeys,
+  filtrerOppgaver,
+  OppgavestatusFilter,
+  OppgavestatusFilterKeys,
+  OppgavetypeFilter,
+  OppgavetypeFilterKeys,
+  SaksbehandlerFilter,
+  SaksbehandlerFilterKeys,
+  YtelseFilter,
+  YtelseFilterKeys,
+  EnhetFilter,
+} from '~components/nyoppgavebenk/Oppgavelistafiltre'
+
+const FilterFlex = styled.div`
+  display: flex;
+  justify-content: space-evenly;
+`
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  margin: 2rem 2rem 2rem 0rem;
+  max-width: 20em;
+  button:first-child {
+    margin-right: 1rem;
+  }
+`
+
+export const Oppgavelista = (props: { oppgaver: ReadonlyArray<OppgaveDTOny>; hentOppgaver: () => void }) => {
+  const { oppgaver, hentOppgaver } = props
+
+  const [saksbehandlerFilter, setSaksbehandlerFilter] = useState<SaksbehandlerFilterKeys>('visAlle')
+  const [enhetsFilter, setEnhetsFilter] = useState<EnhetFilterKeys>('visAlle')
+  const [ytelseFilter, setYtelseFilter] = useState<YtelseFilterKeys>('visAlle')
+  const [oppgavestatusFilter, setOppgavestatusFilter] = useState<OppgavestatusFilterKeys>('visAlle')
+  const [oppgavetypeFilter, setOppgavetypeFilter] = useState<OppgavetypeFilterKeys>('visAlle')
+
+  const mutableOppgaver = oppgaver.concat()
+  const filtrerteOppgaver = filtrerOppgaver(
+    enhetsFilter,
+    saksbehandlerFilter,
+    ytelseFilter,
+    oppgavestatusFilter,
+    oppgavetypeFilter,
+    mutableOppgaver
+  )
 
-export const Oppgavelista = (props: { oppgaver: ReadonlyArray<OppgaveDTOny> }) => {
-  const { oppgaver } = props
   return (
-    <div>
+    <>
+      <FilterFlex>
+        <Select
+          label="Saksbehandler"
+          value={saksbehandlerFilter}
+          onChange={(e) => setSaksbehandlerFilter(e.target.value as SaksbehandlerFilterKeys)}
+        >
+          {Object.entries(SaksbehandlerFilter).map(([key, beskrivelse]) => (
+            <option key={key} value={key}>
+              {beskrivelse}
+            </option>
+          ))}
+        </Select>
+
+        <Select label="Enhet" value={enhetsFilter} onChange={(e) => setEnhetsFilter(e.target.value as EnhetFilterKeys)}>
```

Comment:
Litt fristande med ein komponent for desse select-ane, men veit ikkje om det blir veldig mange parametre?

Reply:
Du sier noe, jeg tror det minimal gain og du må ha generiske typer noe jeg ikke vil ha her.